s3tests_boto3/functional/test_iam.py

```diff
@@ -413,3 +419,304 @@ def test_delete_user_policy_from_multiple_policies():
     client.delete_user_policy(PolicyName='AllowAccessPolicy2', UserName=get_tenant_user_id())
     client.get_user_policy(PolicyName='AllowAccessPolicy3', UserName=get_tenant_user_id())
     client.delete_user_policy(PolicyName='AllowAccessPolicy3', UserName=get_tenant_user_id())
+
+
+@attr(resource='user-policy')
+@attr(method='s3 Actions')
+@attr(operation='Verify Allow Bucket Actions in user Policy')
+@attr(assertion='succeeds')
+@attr('user-policy')
+def test_allow_bucket_actions_in_user_policy():
+    client = get_tenant_iam_client()
+    bucket = get_new_bucket()
+    bucket_listed = False
+
+    policy_document_allow = json.dumps(
+        {"Version": "2012-10-17",
+         "Statement": {
+             "Effect": "Allow",
+             "Action": ["s3:ListAllMyBuckets", "s3:DeleteBucket"],
+             "Resource": "arn:aws:s3:::*"}}
+    )
+    client.put_user_policy(PolicyDocument=policy_document_allow, PolicyName='AllowAccessPolicy',
+                           UserName=get_tenant_user_id())
+    s3_client = get_client()
+    response = s3_client.list_buckets()
+    for index in range(len(response['Buckets'])):
+        if bucket == response['Buckets'][index]['Name']:
+            bucket_listed = True
+            break
+        else:
+            continue
+    if not bucket_listed:
+        raise AssertionError("bucket is not listed")
+
+    s3_client.delete_bucket(Bucket=bucket)
+    response = s3_client.list_buckets()
+    bucket_listed = False
+    for index in range(len(response['Buckets'])):
+        if bucket == response['Buckets'][index]['Name']:
+            bucket_listed = True
+            break
+        else:
+            continue
+    if bucket_listed:
+        raise AssertionError("deleted bucket is getting listed")
+    client.delete_user_policy(PolicyName='AllowAccessPolicy', UserName=get_tenant_user_id())
+
+
+@attr(resource='user-policy')
+@attr(method='s3 Actions')
+@attr(operation='Verify Deny Bucket Actions in user Policy')
+@attr(assertion='succeeds')
+@attr('user-policy')
+def test_deny_bucket_actions_in_user_policy():
```

Comment:
For a user can we add a scenario which allow bucket action, subsequently deny bucket action for same resource and allow bucket action at last and cleanup.

Reply:
This is getting covered in other tests
please check this PR https://github.com/Seagate/s3-tests/pull/4